### PR TITLE
Comply with PEP518

### LINKED ISSUE #108: PEP518 compliance, add a pyproject.toml
For PEP518 compliance we should add a pyproject.toml that specify the build-time dependencies of membrane-curvature.

### PR DESCRIPTION
**Pull request by ojeda-e**
## Description
Fixes #108 and #104  
Add `pyproject,toml` to comply with PEP518. Deleted unnecessary files:
- [.lgtm.yml](https://github.com/MDAnalysis/membrane-curvature/blob/main/.lgtm.yml)
- [_config.yml](https://github.com/MDAnalysis/membrane-curvature/blob/main/_config.yml).

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by ojeda-e** [comment]
Hi @IAlibay, @hmacdope 
Apologies for the slow response here. I added an initial version of the `pyproject.toml` for your review. I added some comments below since I wasn't sure the changes here included are the ones that are expected to comply with the MDAkits.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+requires = ["setuptools", "wheel", "versioneer"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "MembraneCurvature"
+version = "1.0.0"
+license = {file = "LICENSE"}
+description = "MDAnalysis tool to calculate membrane curvature from MD simulations."
+authors = [
+    {name = "Estefania Barreto-Ojeda", email = "estefania@ojeda-e.com"},
```
Comment (ojeda-e) [inline pyproject.toml:11]: Not sure if relevant. Here in `authors`, should I add more names from the MDA core devs?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        'Topic :: Scientific/Engineering ',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12'
+]
+
+[project.optional-dependencies]
+tests = [
```
Comment (ojeda-e) [inline pyproject.toml:45]: For simplicity, I split the dependencies into two groups: tests and docs. If there is a better approach, please let me know.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,65 @@
+[build-system]
+requires = ["setuptools", "wheel", "versioneer"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "MembraneCurvature"
+version = "1.0.0"
```
Comment (IAlibay) [inline pyproject.toml:7]: Because versioneer (or alternative) is handling the versioning, this should be put under the `dynamic` entry instead.

```suggestion
dynamic = ["version"]
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,65 @@
+[build-system]
+requires = ["setuptools", "wheel", "versioneer"]
```
Comment (ojeda-e) [inline pyproject.toml:2]: I had to add `versioneer` here. I'd rather skip it, but found out that w/o it the wheel doesn't build.

Comment (IAlibay) [inline pyproject.toml:2]: Mostly a todo for myself - I'm not sure you need to/should include versioneer in the build requirements if versioneer is already vendored.

Maybe as an alternative to versioneer - we've been moving a lot of stuff to versioningit instead (see https://github.com/MDAnalysis/PathSimAnalysis/pull/12 for an example). It gets rid of some messy toml dependencies whilst also avoiding vendoring things. Note: the real reason we're moving away from versioneer is that it hasn't been as consistently maintained lately, you could stay with versioneer for now without issues though.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+[project]
+name = "MembraneCurvature"
+version = "1.0.0"
+license = {file = "LICENSE"}
+description = "MDAnalysis tool to calculate membrane curvature from MD simulations."
+authors = [
+    {name = "Estefania Barreto-Ojeda", email = "estefania@ojeda-e.com"},
+]
+maintainers = [
+    {name = 'MDAnalysis Core Developers', email = 'mdanalysis@numfocus.org'}
```
Comment (IAlibay) [inline pyproject.toml:14]: Pinging @orbeckst due to previous discussions on the topic - I'm not 100% sure if the coredevs can take on full maintenance right now (maybe they can, but I could not make that decision for them).

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+dependencies = [
+    'numpy>=1.20.0',
+    'mdanalysis>=2.0.0'
+    ]
+
+classifiers = [
+        'Development Status :: 5 - Production/Stable',
+        'Environment :: Console',
+        'Intended Audience :: Science/Research',
+        'License :: OSI Approved :: GNU General Public License v3 (GPLv3)',
```
Comment (IAlibay) [inline pyproject.toml:32]:
```suggestion
        'License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)',
```
Apologies, I can't remember if this repo is GPLv3+ or GPLv3 pure

File: pyproject.toml (hunk 2 above)
Comment (IAlibay) [reply pyproject.toml:45]: That sounds good to me, it's very much the standard that other packages also follow :)

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+tests = [
+    "pytest",
+    "pytest-cov",
+    "ipython",
+    "MDAnalysisTests"
+]
+docs = [
+    "ipython",
+    "MDAnalysisTests",
+    "sphinx_rtd_theme",
```
Comment (IAlibay) [inline pyproject.toml:54]: todo: there's an MDAnalysis sphinx theme that we've been moving folks over to (it's more up to date) - maybe we can do that as a separate PR?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Apologies it's a very very quick review (it's quite late on my end 😅 ), but here are some initial thoughts.

Overall this looks great, there's a few things I would advise changing, the big ones (e.g. versioningit) are very much lightweight suggestions.

Main thing that would need changing here is the `dynamic` entry for `version`.

File: pyproject.toml (hunk 4 above)
Comment (IAlibay) [reply pyproject.toml:2]: Ah did not see that comment before my review, sorry!

File: pyproject.toml (hunk 7 above)
Comment (ojeda-e) [reply pyproject.toml:54]: yes, good idea. See #120

File: pyproject.toml (hunk 6 above)
Comment (ojeda-e) [reply pyproject.toml:32]: Thanks for checking this, it's GPLv3 (with no plus). 100% positive.

File: pyproject.toml (hunk 4 above)
Comment (IAlibay) [reply pyproject.toml:2]: Ok I've worked out what's going wrong with the whole `versioneer` needing to be installed thing.

Turns out that it needs additional stuff for the vendored file to be picked up: https://github.com/python-versioneer/python-versioneer/issues/249

The diff would need to look something like this:

```
diff --git a/pyproject.toml b/pyproject.toml
index d1ed562..6249d7b 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools", "wheel", "versioneer"]
+requires = ["setuptools", "wheel", 'tomli; python_version < "3.11"']
 build-backend = "setuptools.build_meta"
 
 [project]
@@ -62,4 +62,4 @@ include = ["membrane_curvature*"]
 
 [project.urls]
 Documentation = "https://membrane-curvature.readthedocs.io/en/latest/"
-Changelog = "https://github.com/MDAnalysis/membrane-curvature/blob/main/CHANGELOG.rst"
\ No newline at end of file
+Changelog = "https://github.com/MDAnalysis/membrane-curvature/blob/main/CHANGELOG.rst"
diff --git a/setup.py b/setup.py
index e4e8d2e..0880925 100644
--- a/setup.py
+++ b/setup.py
@@ -2,8 +2,12 @@
 MembraneCurvature MDAkit
 A tool to calculate Membrane Curvature
 """
+import os
 import sys
 from setuptools import setup, find_packages
+
+sys.path.append(os.path.dirname(__file__))
+
 import versioneer
 
 short_description = __doc__.split("\n")
```

Comment (IAlibay) [reply pyproject.toml:2]: @ojeda-e at this point I would suggest just switching over to versioningit, it'll make life a lot easier.

Comment (IAlibay) [reply pyproject.toml:2]: I've opened a PR into this PR with the necessary changes.

Comment (ojeda-e) [reply pyproject.toml]: Thanks @IAlibay!

**Comment by IAlibay** [comment]
Oops not sure what happened with the tests, sorry! I'll try to have a stab at fixing up whatever I messed up a bit later today.

**Comment by ojeda-e** [comment]
No worries, thanks for your help here. I think it's just a missing package for the metadata when running tests. I'll give a try.

**Comment by IAlibay** [comment]
@ojeda-e looks like the package name internally is "MembraneCurvature" instead of "membrane_curvature" is that intended?

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,95 @@
+[build-system]
+requires = ["setuptools", "versioningit"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "MembraneCurvature"
```
Comment (IAlibay) [inline pyproject.toml:6]:
```suggestion
name = "membrane_curvature"
```
This should do the trick I think

**Review by IAlibay — APPROVED** [review]
So overall from my end, i.e. "code wise", this looks good so I'm going to approve. However, I don't have a clue about the whole authors & maintainers stuff.

I think it's something to be discussed once a few of the coredevs come back from their holidays.

**Comment by ojeda-e** [comment]
Sounds good, thanks for your help and pointers here @IAlibay 
Will just reverse the line I added and it will be ready for coredevs to review.

File: pyproject.toml (hunk 5 above)
Comment (orbeckst) [reply pyproject.toml:14]: We can't promise maintaining the kit. Therefore we need to find another name here. 

In the absence of a maintainer, I suggest putting @ojeda-e back in there. Then there's at lease contact information and it's the historical status. I don't know what the best way is to indicate "looking for maintainer".

File: pyproject.toml (hunk 1 above)
Comment (orbeckst) [reply pyproject.toml:11]: Your name should definitely be there. If there are other major contributors you can add them. You might be able to reference a AUTHORS file — check the syntax for pyproject.toml.

**Review by orbeckst — CHANGES_REQUESTED** [review]
The MDAnalysis devs can't promise maintaining the kit. Therefore we need to find another name/email for `maintainers`.

If you're looking for a new maintainer then I would 
- put your own name back in maintainer
- add a conspicuous note in README and in the docs saying that you're looking for a maintainer and how to contact

File: pyproject.toml (hunk 5 above)
Comment (ojeda-e) [reply pyproject.toml:14]: Thanks @orbeckst.
I will add my name as maintainer until there is a candidate.

File: pyproject.toml (hunk 1 above)
Comment (ojeda-e) [reply pyproject.toml:11]: Thnaks @orbeckst 
I added the AUTHORS file, however, the `toml` file doesn't like the syntax `authors = { file = "AUTHORS" }`. I checked and the wheel is not generated:

```
      {"file":"AUTHORS"} is not of type "array"
      ValueError: invalid pyproject.toml config: `project.authors`.
      configuration error: `project.authors` must be array
```
Can't find any example that parses a file. According to [setuptools docs](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration), the AUTHORS file is added by default in the `license-files` key.  I checked and it is the file is added to the wheel. Here ta snippet of the wheel contents:
```
      292  01-02-2024 22:50   membrane_curvature-1.0.0+16.g5d02d31.dist-info/AUTHORS
    34684  01-02-2024 22:50   membrane_curvature-1.0.0+16.g5d02d31.dist-info/LICENSE
    47817  01-02-2024 22:50   membrane_curvature-1.0.0+16.g5d02d31.dist-info/METADATA
```

**Comment by ojeda-e** [comment]
With [5d02d31](https://github.com/MDAnalysis/membrane-curvature/pull/119/commits/5d02d3174d2cd326ce93568bd49d2f25da91b7f3) this PR would also close #117

File: AUTHORS [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+Chronological list of authors
+-----------------------------
+
+2021
+  - Estefania Barreto-Ojeda
+  - Oliver Beckstein
+  - Irfan Alibay
+
+2023
+  - Lily Wang
```
Comment (ojeda-e) [inline AUTHORS:14]: @IAlibay you contributed in 2023, should I repeat your name here as well?

File: README.md [hunk 10]
```diff
@@ -11,6 +11,8 @@ Membrane Curvature
 MembraneCurvature is an [MDAnalysis] tool to calculate membrane curvature from 
 Molecular Dynamics simulations. 
 
+> **Interested in becoming a maintainer?** We welcome your passion and expertise to help shape and grow this open-source project! Please contact estefania@ojeda-e.com for more details.
+
```
Comment (ojeda-e) [inline README.md:15]: @orbeckst I added this comment to look for maintainers. Should I add an MDAnalysis email better?

File: AUTHORS (hunk 9 above)
Comment (IAlibay) [reply AUTHORS:14]: Usually the easiest thing is for author files to be "at first time of contribution", that makes life a ton easier for maintenance going forward 😅

Comment (ojeda-e) [reply AUTHORS:14]: Gotcha, thanks for confirming.

File: README.md (hunk 10 above)
Comment (orbeckst) [reply README.md:15]: If I were someone interested in maintaining, _you_ would be the person I'd most like to talk to — so I think having you there is best, provided that you'll look at emails.

If you get in contact with someone and need advice or just talk things through then just get in touch with the MDA team. (One advantage you have over many other people is that you know who to talk to because you worked with the MDAnalysis project!)

**Review by orbeckst — APPROVED** [review]
I am happy with the changes.

(Sorry, I don't know if/how AUTHORS files work in the authors variable, I just made a random guess.)

File: pyproject.toml (hunk 1 above)
Comment (IAlibay) [reply pyproject.toml:11]: Yeah as far as I'm aware authors can be a list of dictionaries but not a file. Personally, I would stick with yourself here. It's good enough to have an `AUTHORS` file.

**Review by IAlibay — APPROVED** [review]
> @IAlibay would please help me with reviewing AUTHORS? Hopefully I didn't miss anything there,

Quickly scanning through there was one PR by a now deleted Github user - just docstring changes.

I'd say the authors file looks good to me!
